python/ray/train/_internal/data_config.py

```diff
@@ -82,6 +82,10 @@ def configure(
         """
         output = [{} for _ in range(world_size)]
 
+        for dataset_name, dataset in datasets.items():
+            if dataset.name is None:
+                dataset.set_name(dataset_name)
```

Comment:
Small question here. How does this name get used by the individual dataset splits (the output of `ds.streaming_split()`? Will each split inherit the same name? Will they have the same name + some split suffix?